src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1665,6 +1665,10 @@ SubqueryPushdownMultiNodeTree(Query *originalQuery)
 	 * expression on the LIMIT and OFFSET clauses. Note that logical optimizer
 	 * expects those clauses to be already evaluated.
 	 */
+	extendedOpNode->originalLimitCount =
+		PartiallyEvaluateExpression(extendedOpNode->originalLimitCount, NULL);
+	extendedOpNode->originalLimitOffset =
+		PartiallyEvaluateExpression(extendedOpNode->originalLimitOffset, NULL);
```

Comment:
PartiallyEvaluateExpression doesn't seem like something we should be doing in the planner, since it will evaluate stable and volatile functions, which can even write to the database.

Reply:
Hmz, I see we do this below, that's a bad idea too. There should at least be a check here to confirm that there are no stable or volatile functions (`contain_mutable_functions`) before we do execution.

Reply:
Removed partially evaluating original limit/offset. Updating to not `PartiallyEvaluateExpression` at all here would be a separate PR